src/Microsoft.DotNet.Build.Tasks/PackageFiles/init-tools.cmd

```diff
@@ -0,0 +1,34 @@
+@echo off
+setlocal
+
+set PROJECT_DIR=%~1
+set DOTNET_CMD=%~2
+set TOOLRUNTIME_DIR=%~3
+set TOOLS_DIR=%~dp0
+
+if not exist "%PROJECT_DIR%" (
+  echo ERROR: Cannot find project root path at [%PROJECT_DIR%]. Please pass in the source directory as the 1st parameter.
+  exit /b 1
+)
+
+if not exist "%TOOLS_DIR%" (
+  echo ERROR: Cannot find tools path at [%TOOLS_DIR%]. Please pass in the tools directory as the 2nd parameter.
+  exit /b 1
+)
+
+if not exist "%DOTNET_CMD%" (
+  echo ERROR: Cannot find dotnet command at [%DOTNET_CMD%]. Please pass in the dotnet command as the 3rd parameter.
+  exit /b 1
+)
+
+set tempDirectory=%cd%
+cd %TOOLS_DIR%\tool-runtime\
+call %DOTNET_CMD% restore
+call %DOTNET_CMD% publish -f dnxcore50 -r win7-x64 -o %TOOLRUNTIME_DIR%
+cd %tempDirectory%
+
+xcopy /E /Y /Q %TOOLS_DIR%..\lib %TOOLRUNTIME_DIR%
+
+call type NUL > %TOOLRUNTIME_DIR%\ToolRuntime.semaphore
```

Comment:
Didn't we decide the semaphore was going to have the version and be handled in the individual repo scripts?